src/ppx/application.cpp

```diff
@@ -1142,55 +1142,14 @@ bool Application::IsRunning() const
     return mWindow->IsRunning();
 }
 
-int Application::Run(int argc, char** argv)
+void Application::FinalizeSettings()
 {
-    // Only allow one instance of Application. Since we can't stop
-    // the app in the ctor - stop it here.
-    //
-    if (this != sApplicationInstance) {
-        return false;
-    }
-
-    // Parse args.
-    if (Failed(mCommandLineParser.Parse(argc, const_cast<const char**>(argv)))) {
-        PPX_ASSERT_MSG(false, "Unable to parse command line arguments");
-        return EXIT_FAILURE;
-    }
-
-    // Call config.
-    // Put this early because it might disable the display.
-    DispatchConfig();
-
-    // Knobs need to be set up before commandline parsing.
-    // This has dependency on DispatchConfig() where standard knob default values are set
-    DispatchInitKnobs();
-
-    if (!mKnobManager.IsEmpty()) {
-        mCommandLineParser.AppendUsageMsg(mKnobManager.GetUsageMsg());
-    }
-
-    if (mCommandLineParser.GetOptions().GetOptionValueOrDefault("help", false)) {
-        PPX_LOG_INFO(mCommandLineParser.GetUsageMsg());
-        return EXIT_SUCCESS;
-    }
-
-    // Command line will overwrite the knob settings from the application
-    if (!mKnobManager.IsEmpty()) {
-        auto options = mCommandLineParser.GetOptions();
-        mKnobManager.UpdateFromFlags(options);
-    }
-
-    // Asset directories based on settings in mSettings, mCommandLineParser and mKnobManager
-    // note that mKnobManager needs to be updated by options from mCommandLineParser before this call
-    UpdateAssetDirs();
-
     if (mSettings.appName.empty()) {
         mSettings.appName = "PPX Application";
     }
     if (mSettings.window.title.empty()) {
         mSettings.window.title = mSettings.appName;
```

Comment:
Same here, could the default constructor also initialize this to kDefaultAppName and those lines removed?

Reply:
updated